tests/integration/test_charm.py

```diff
@@ -0,0 +1,147 @@
+            channel="edge",
+            num_units=1,
+            series="bionic",
+        ),
+    )
+    # Attach resources to charms.
+    await ops_test.juju("attach-resource", SLURMCTLD, f"etcd={slurmctld_res['etcd']}")
+    await ops_test.juju("attach-resource", SLURMD, f"nhc={slurmd_res['nhc']}")
+    # Set integrations for charmed applications.
+    await ops_test.model.relate(f"{SLURMCTLD}:{SLURMD}", f"{SLURMD}:{SLURMD}")
+    await ops_test.model.relate(f"{SLURMCTLD}:{SLURMDBD}", f"{SLURMDBD}:{SLURMDBD}")
+    await ops_test.model.relate(f"{SLURMDBD}:db", f"{DATABASE}:db")
+    # Reduce the update status frequency to accelerate the triggering of deferred events.
+    async with ops_test.fast_forward():
+        await ops_test.model.wait_for_idle(apps=[SLURMCTLD], status="active", timeout=1000)
+        assert ops_test.model.applications[SLURMCTLD].units[0].workload_status == "active"
+
+
+@pytest.mark.abort_on_fail
+@tenacity.retry(
+    wait=tenacity.wait.wait_exponential(multiplier=2, min=1, max=30),
+    stop=tenacity.stop_after_attempt(3),
+    reraise=True,
+)
+async def test_slurmctld_is_active(ops_test: OpsTest) -> None:
+    """Test that slurmctld is active inside Juju unit."""
+    logger.info("Checking that slurmctld is active inside Juju unit...")
+    async with unit_connection(ops_test, SLURMCTLD, UNIT) as conn:
+        stdin, stdout, stderr = conn.exec_command("systemctl is-active slurmctld")
+        assert codecs.decode(stdout.read()).strip("\n") == "active"
+
+
+@pytest.mark.abort_on_fail
+@tenacity.retry(
+    wait=tenacity.wait.wait_exponential(multiplier=2, min=1, max=30),
+    stop=tenacity.stop_after_attempt(3),
+    reraise=True,
+)
+async def test_slurmctld_port_listen(ops_test: OpsTest) -> None:
+    """Test that slurmctld is listening on port 6817."""
+    logger.info("Checking that slurmctld is listening on port 6817...")
+    async with unit_connection(ops_test, SLURMCTLD, UNIT) as conn:
+        stdin, stdout, stderr = conn.exec_command("sudo lsof -i -n | grep ':6817'")
+        assert "LISTEN" in codecs.decode(stdout.read())
+
+
+@pytest.mark.abort_on_fail
+@tenacity.retry(
+    wait=tenacity.wait.wait_exponential(multiplier=2, min=1, max=30),
+    stop=tenacity.stop_after_attempt(3),
+    reraise=True,
+)
+async def test_etcd_is_active(ops_test: OpsTest) -> None:
+    """Test that slurmctld is active inside Juju unit."""
+    logger.info("Checking that etcd is active inside Juju unit...")
+    async with unit_connection(ops_test, SLURMCTLD, UNIT) as conn:
+        stdin, stdout, stderr = conn.exec_command("systemctl is-active etcd")
+        assert codecs.decode(stdout.read()).strip("\n") == "active"
+
+
+@pytest.mark.abort_on_fail
+@tenacity.retry(
+    wait=tenacity.wait.wait_exponential(multiplier=2, min=1, max=30),
+    stop=tenacity.stop_after_attempt(3),
+    reraise=True,
+)
+async def test_munge_is_active(ops_test: OpsTest) -> None:
+    """Test that slurmctld is active inside Juju unit."""
```

Comment:
Looks like this needs to be updated for munge

Reply:
Rookie mistake :clown_face:

Reply:
Resolved in https://github.com/canonical/slurmctld-operator/pull/1/commits/975e4a3d1548d6df577d08572406eb25c07639d9